pkg/daemon/manager/manager.go

```diff
@@ -31,14 +33,26 @@ func (m *manager) CreateService(name, exepath string, config mgr.Config, args ..
 	service, err := underlyingMgr.CreateService(name, exepath, config, args...)
 	return winsvc.Service(service), err
 }
+
 func (m *manager) ListServices() ([]string, error) {
 	underlyingMgr := (*mgr.Mgr)(m)
 	return underlyingMgr.ListServices()
 }
+
 func (m *manager) OpenService(name string) (winsvc.Service, error) {
 	underlyingMgr := (*mgr.Mgr)(m)
 	return underlyingMgr.OpenService(name)
 }
+
+func (m *manager) CloseService(name string) error {
+	underlyingMgr := (*mgr.Mgr)(m)
+	winSvc, err := underlyingMgr.OpenService(name)
+	if err != nil {
+		return err
+	}
+	return winSvc.Close()
+}
```

Comment:
This implementation doesn't make sense. you're opening a handle which won't be used anywhere only to close the specific handle.

In general, I also disagree with moving this close method to the manager at all.
In go handles are closed via a method on the handle itself.
See examples in the `os` and `io` standard libraries:
* [Pipes](https://pkg.go.dev/io#PipeReader.Close)
* [Files](https://pkg.go.dev/os#File.Close)

Reply:
Good point @sebsoto. I agree with you. I obviously had forgotten how file and pipe handles closing work.

@saifshaikh48 sorry for not thinking this through and catching it sooner.